Update branding of Smaragd

benruehl opened the pull request:
## Summary
This updates the project's logo and associated files and reworks its public presentation.

## Motivation and Context
The current logo looks a little outdated because it is too detailed. It is also a little boring because it lacks visual diversion. The readme could be more inviting to visitors and express more detailed how the library can be used.

## Details
- [x] Update logo
- [ ] Update readme
   - [x] Update *Features* section - more interesting visual presentation, clearer structure
   - [x] Update *Quick Start* section - more content and code snippets to see how the library feels like
   - [ ] Update *Why another library* section - be more clear about differences to similar libraries
- [x] Remove existing logo files

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] ~~My code follows the code style of this project.~~
- [ ] My change requires a change to the documentation.
- [ ] ~~I have updated the documentation accordingly.~~
- [ ] ~~I have written/altered unit tests for the changes.~~
- [x] Only necessary files have been added/altered.

benruehl committed:
Update logo in readme

benruehl committed:
Update 'features' section of readme

Add a class diagram showing the core features of the library.
Reorganize the textual content to fit to the image.

benruehl committed:
Update 'quick start' section of readme

Replace the existing instructions and links with a code demonstration.
The existing instructions don't transport enough knowledge to actually start
with development. A link to another complex project to look into is kind of
demotivating for first time readers.

nkristek reviewed (dismissed):
Great work, I like the dependency graph a lot!

benruehl committed:
Update icons and social preview

Update resource files like the icon of Smaragd and the social preview image to the new theme. Replace the various
Affinity Designer files with a single file that contains all resources.

nkristek reviewed (approved):
LGTM, thanks for your contribution!
